### PR TITLE
✨ Feature: Auto-Link URLs in Post Text and Comments (Closes #122)

### LINKED ISSUE #122: ✨Feature: Auto-Link URLs in Comments Or Chats
### ✨ Feature Request: Auto-Link URLs in Comments Or Chats

**Summary**  
When a user includes a URL in post text (and comments/quotes), it should automatically render as a clickable hyperlink. Links must open in a new tab and be safe (sanitized) across desktop and mobile.

---

**User Story**  
As a reader, I want URLs in posts to be clickable so I can quickly visit referenced sources without copy-pasting.

---

**Acceptance Criteria**  
- [ ] Detect and hyperlink valid URLs in post body text (e.g., `https://…`, `http://…`, and common `www.` patterns).  
- [ ] Hyperlinks open in a new tab and use `rel="noopener noreferrer"` for security.  
- [ ] Styling matches our text/link guidelines (visible focus state, accessible contrast).  
- [ ] Works consistently in **posts**, **comments**, **quotes/excerpts**, **activity cards**, and **search results snippets**.  
- [ ] Long URLs are truncated visually (e.g., with ellipsis) but preserve the full href on click and in title tooltip.  
- [ ] Input is sanitized to prevent XSS; only safe protocols are allowed (`http`, `https`).  
- [ ] Tappable on mobile with adequate touch target sizing.

---

**Tech Notes**  
- Consider using a robust linkification utility (e.g., `linkify-it`, `autolinker`, or MUI-compatible rich text renderer) applied at render time.  
- Add unit tests for edge cases: punctuation at end of URL, parentheses, query strings, and unicode domains.  
- Ensure server-side rendering (if applicable) produces the same markup to support link unfurling/crawlers.

---

**Priority**  
Low – Improves readability and credibility by making references instantly usable.

### PR DESCRIPTION
**Pull request by PriyanshuNaik27**
# Description

This PR adds automatic URL linking functionality to the QuoteVote application. When users post content containing URLs, they will now be automatically converted to clickable links, improving user experience and making shared content more interactive.

Implements auto-link URL detection and conversion functionality. This feature scans text content for URL patterns and automatically converts them to HTML anchor tags, allowing users to click on URLs without manual formatting.

#122

### CODE REVIEW

**Comment by flyblackbox** [comment]
Hello @PriyanshuNaik27 I am not seeing the hyperlink when checking this post below.
Will it only work moving forward?

https://deploy-preview-137--quotevote.netlify.app/post/singularity/the-gentle-singularity-%E2%80%8Bby-sam-altman/e_6A4M

image

I also tested by creating a new post but it also did not work as expected.
https://deploy-preview-137--quotevote.netlify.app/post/qa/qa-testing-url-link/68dec2e570299200263b1c7f

**Comment by flyblackbox** [comment]
@PriyanshuNaik27 we need better documentation on how to do this.  Many devs are asking.  

Do you think you can figure it out and update our ReadMe?  If so, please create an issue and I will add to the roadmap and assign you. 

Otherwise, @olivermolina or @niamao might be able to help you.

File: client/package.json [hunk 1]
```diff
@@ -101,7 +103,7 @@
     "build": "ROLLUP_SKIP_NATIVE=true vite build && gulp licenses",
     "build:ci": "ROLLUP_SKIP_NATIVE=true NODE_ENV=production vite build && gulp licenses",
     "preview": "vite preview",
-    "test": "vitest -u -w 1 --env=jsdom --silent",
+    "test": "vitest -u -w 1  --silent",
```
Comment (niamao) [inline client/package.json:106]: why was this part removed ?

File: package-lock.json
Comment (niamao) [inline package-lock.json:1]: do we need to include lock file in our PR? @olivermolina

**Review by niamao — CHANGES_REQUESTED** [review]

File: client/package.json (hunk 1 above)
Comment (flyblackbox) [reply client/package.json]: Ah OK.  Thank you!  Many are having problems with local testing that involves the database.  We are working on explaining why and a solution.
